src/listener/keyboard_listener_provider.rs

```diff
@@ -0,0 +1,7 @@
+use std::error::Error;
+
+pub trait KeyboardListenerProvider: Sized {
+    fn new(callback: fn()) -> Result<Self, Box<dyn Error>>;
```

Comment:
if this is a generic KeyboardListener why do I only give the callback and not an option to supply what I want to listen for? -> this will come in handy when we switch from Ctrl+C to Shift+Ctrl+C